include/mpp/algorithm/back_substitution.hpp

```diff
@@ -93,10 +111,27 @@ namespace mpp
 				}
 			}
 
-			return matrix<To, RowsExtent, ColumnsExtent>{ a.rows(),
-				1,
-				back_subst_on_buffer<To, RowsExtent, ColumnsExtent, Check>(a.data(), b.data(), a.rows()),
-				unsafe };
+			const auto rows = a.rows();
+
+			using x_mat_t = matrix<To, RowsExtent, ColumnsExtent, ToAllocator>;
+			auto&& x_buf  = back_subst_on_buffer<Check,
+                RowsExtent,
+                ColumnsExtent,
+                typename std::allocator_traits<ToAllocator>::template rebind_alloc<To>>(a.data(),
```

Comment:
The allocator should be rebinded to `default_floating_type`
```suggestion
                typename std::allocator_traits<ToAllocator>::template rebind_alloc<default_floating_type>>(a.data(),
```